packages/image_picker/image_picker_platform_interface/lib/src/types/multi_image_picker_options.dart

```diff
@@ -6,11 +6,31 @@ import 'image_options.dart';
 
 /// Specifies options for picking multiple images from the device's gallery.
 class MultiImagePickerOptions {
-  /// Creates an instance with the given [imageOptions].
+  /// Creates an instance with the given [imageOptions] and [limit].
   const MultiImagePickerOptions({
     this.imageOptions = const ImageOptions(),
+    this.limit,
   });
 
+  /// Creates an instance with the given [imageOptions] and [limit].
+  /// Throws if limit is not valid.
+  MultiImagePickerOptions.createAndValidate({
+    this.imageOptions = const ImageOptions(),
+    this.limit,
+  }) {
+    _validateOptions(limit: limit);
+  }
+
   /// The image-specific options for picking.
   final ImageOptions imageOptions;
+
+  /// The maximum number of images to select. Default null value means no limit.
+  final int? limit;
+
+  /// Validates that all values are within required ranges. Throws if not.
```

Comment:
Both of these comments need to be broken up into multi line comments by sentence. (check the style guide I linked earlier)